### PR TITLE
iOS: Fixes #6318: Beta Editor: Remove white border

### LINKED ISSUE #6318: iOS Beta Editor: White border around editor
## Environment

Joplin version: 
Platform: Mobile/iOS
OS specifics: iOS 15.4

## Steps to reproduce

1. Switch to a dark theme
2. Enable the beta editor
3. Edit a note

## Describe what you expected to happen

The keyboard should be in dark mode and there should be no white border around the editor.

## Screenshot

White border around page content in dark mode

### PR DESCRIPTION
**Pull request by personalizedrefrigerator**
# Summary
This **should** fix #6318.

I can't easily deploy to my iOS device (so I'm testing this on Android). It would be very helpful if someone could confirm that this does indeed fix #6318!

Thank you!

### CODE REVIEW

**Comment by laurent22** [comment]
I can't replicate this on Android (for me the border is dark as expected), so I'm not sure about this fix. I'd prefer if someone could confirm they have this issue too.

**Comment by personalizedrefrigerator** [comment]
> I can't replicate this on Android (for me the border is dark as expected), so I'm not sure about this fix. I'd prefer if someone could confirm they have this issue too.

I'm **only experiencing the issue on iOS**. I think it's because the WebView has a transparent background on Android that we don't see the border. **Edit:** This _might_ also fix #5775.

File: packages/app-mobile/components/NoteEditor/NoteEditor.tsx [hunk 1]
```diff
@@ -243,8 +257,8 @@ function NoteEditor(props: Props, ref: any) {
 		}
 	`;
 
-	// const css = useCss(props.themeId);
-	const html = useHtml('');
+	const css = useCSS(props.themeId);
```
Comment (laurent22) [inline packages/app-mobile/components/NoteEditor/NoteEditor.tsx]: You don't need an effect. When a variable value is simply derived from another variable, you can use useMemo. Just give props.themeId as dependency.

Comment (personalizedrefrigerator) [reply packages/app-mobile/components/NoteEditor/NoteEditor.tsx]: That should be changed!

File: packages/app-mobile/components/NoteEditor/NoteEditor.tsx [hunk 2]
```diff
@@ -169,6 +169,20 @@ function fontFamilyFromSettings() {
...
+	const [css, setCSS] = useState('');
+
+	useMemo(() => {
+		const theme = themeStyle(themeId);
+		setCSS(`
+			:root {
+				background-color: ${theme.backgroundColor};
+			}
+		`);
+	}, [themeId]);
```
Comment (laurent22) [inline packages/app-mobile/components/NoteEditor/NoteEditor.tsx]: Your code is correct but it does more than what is needed. If all you need is a derived value, the pattern is to do `return useMemo(() .....`, then inside the function `return ":root { ...."`. Then React will take of handling the state and so on. You don't need to manage state for this basically.

In fact I wouldn't create a custom effect for this (I'd just do `const css = useMemo...` but that's up to you.

Comment (personalizedrefrigerator) [reply packages/app-mobile/components/NoteEditor/NoteEditor.tsx]: That should now be done!

**Comment by laurent22** [comment]
Thanks for the update!

**Pull request merged**